### ISSUE TITLE
Incorrect class of 'this' object in Java.registerClass()'s interface implementation methods

### ISSUE THREAD

**Issue by gebing** [frida/frida-java#133]
The Java class of `this` object in Java.registerClass()'s interface implementation methods is the class of interface, and is not the newly created class. It is a little bit inconvenient for needing convert `this` to newly created class manually when accessing other methods or fields, such like below:

```
var UserDefinedFields = Java.registerClass({
  name: 're.frida.UserDefinedFields',
  implements: [Java.use('re.frida.Formatter')],
  fields: {
    fieldInt: 'int',
    fieldStr: 'java.lang.String',
  },
  methods: {
    format: [{
      returnType: 'java.lang.String',
      argumentTypes: ['int'],
      implementation: function (val) {
        const self = Java.cast(this, UserDefinedFields);
        return self.fieldInt.value;
      }
    }, {
      returnType: 'java.lang.String',
      argumentTypes: ['java.lang.String'],
      implementation: function (val) {
        const self = Java.cast(this, UserDefinedFields);
        return self.fieldStr.value;
      }
    }]
  }
});
```

**Comment by gebing** [comment]
@oleavr 
I made a fix for this issue and merged into pull request #130.

**Comment by oleavr** [comment]
Thanks!
